packages/dapi/lib/grpcServer/handlers/blockheaders-stream/subscribeToBlockHeadersWithChainLocksHandlerFactory.js

```diff
@@ -103,42 +103,22 @@ function subscribeToBlockHeadersWithChainLocksHandlerFactory(
       subscribeToNewBlockHeaders(mediator, zmqClient, coreAPI);
     }
 
-    // If block height is specified instead of block hash, we obtain block hash by block height
-    if (fromBlockHash === '') {
-      if (fromBlockHeight === 0) {
-        throw new InvalidArgumentGrpcError('minimum value for `fromBlockHeight` is 1');
-      }
-
-      // we don't need to check bestBlockHeight because getBlockHash throws
-      // an error in case of wrong height
-      try {
-        fromBlockHash = await coreAPI.getBlockHash(fromBlockHeight);
-      } catch (e) {
-        if (e.code === -8) {
-          // Block height out of range
-          throw new NotFoundGrpcError('fromBlockHeight is bigger than block count');
-        }
-
-        throw e;
-      }
-    }
-
     let fromBlock;
 
     try {
-      // TODO: rework with getBlockStats
-      fromBlock = await coreAPI.getBlock(fromBlockHash);
+      fromBlock = await coreAPI.getBlockStats(fromBlockHash || fromBlockHeight, ['height']);
     } catch (e) {
       // Block not found
       if (e.code === -5) {
-        throw new NotFoundGrpcError('fromBlockHash is not found');
+        throw new NotFoundGrpcError('Block not found');
```

Comment:
Let's make it more verbose, please. Like "Block {fromBlockHash || fromBlockHeight} not found"